CMakeLists.txt

```diff
@@ -36,7 +36,7 @@ endif()
 # ``-DCMAKE_CXX_STANDARD=<standard>`` when invoking CMake
 set(CMAKE_CXX_STANDARD 17 CACHE STRING "")
 
-if(NOT CMAKE_CXX_STANDARD MATCHES "17|20")
+if(NOT CMAKE_CXX_STANDARD MATCHES "17|20|23")
```

Comment:
:) I like the optimism. I have nothing against adding this here, but we are probably effectively locked to 17 until ROOT supports 20.

Reply:
I think this may be true, to some extent. We use podio in the c++20 framework from EIC, and whatever c++20 support is in the headers is available to our code, even if only c++17 functionality is compiled into the libraries. So, I think that here in the CMakeLists.txt indeed the flag is a bit premature but it was a way to ensure the unit tests get built with c++23 for the zip views :-)

Reply:
I guess that just means we should keep c++20 functionality in headers only.

Reply:
The main issue with ROOT appears to be when doing e.g. `.include ExampleMCCollection.h` for a c++20 header in a c++17 ROOT. But if the c++20 functionality in the header is suitably guarded against non-c++20 compilers, then we should be fine.